server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -116,6 +116,8 @@ public class IndexRequest extends ReplicatedWriteRequest<IndexRequest> implement
 
     private boolean requireDataStream;
 
+    private boolean includeSourceOnError = true;
```

Comment:
same comment as earlier, this would be clearer in the StreamInput ctor

Reply:
IndexRequest also has a couple of different public constructors. If you prefer i can move into each ctor ....

Reply:
made it a nullable Boolean to avoid having to handle defaults in multiple places, with that this should also be resolved
https://github.com/elastic/elasticsearch/pull/120725/commits/6a3e74a53bd4cf01cf75c7f59cc5f8ce1be29aef